iceoryx2-bb/cxx/include/iox2/container/optional.hpp

```diff
@@ -332,22 +332,14 @@ Optional(T) -> Optional<T>;
 
 #endif
 
-#if __cplusplus >= 201703L
-// NOLINTNEXTLINE(readability-identifier-naming), for consistency with C++17 code using std::optional
-inline constexpr NulloptT nullopt { detail::NulloptTConstructorTag {} };
-#else
 namespace detail {
 template <typename = void>
 struct NulloptHelper {
     static NulloptT nullopt;
 };
-template <>
-// NOLINTNEXTLINE(misc-definitions-in-headers), not an ODR violation because of template
-NulloptT NulloptHelper<>::nullopt = NulloptT { detail::NulloptTConstructorTag {} };
```

Comment:
@ComicSansMS I had to move this to a `optional.cpp` since there were `multiple definition` errors in the tests. Feel free to use your magic to revert my workaround :)